executor/join_test.go

```diff
@@ -988,3 +990,52 @@ func (s *testSuite2) TestHashJoin(c *C) {
 	innerExecInfo = row[3][4].(string)
 	c.Assert(innerExecInfo[len(innerExecInfo)-1:], LessEqual, "5")
 }
+
+func (s *testSuite2) TestJoinDecimal(c *C) {
+	// Test joining decimals with different fraction
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test")
+	tk.MustExec("drop table if exists t")
+	tk.MustExec("create table t(a int, b decimal(13,2), c decimal(14,4), index b (b) , index c (c))")
+	tk.MustExec("insert into t value(1, 1.12, 1.12)")
+
+	result := tk.MustQuery("explain select /*+ TIDB_INLJ(t1,t2) */ t1.a from t t1 join t t2 on t1.b = t2.c")
+	rows := result.Rows()
+	joinRow := fmt.Sprintf("%v", rows[1])
+	isIndexJoin := strings.Contains(joinRow, "IndexJoin")
+	c.Assert(isIndexJoin, Equals, true)
+	//| Projection_5             | 1.00  | root | t1.a                                                                        |
+	//| └─IndexJoin_9            | 1.00  | root | inner join, inner:IndexReader_8, outer key:t1.b, inner key:t2.c             |
+	//|   ├─TableReader_11       | 1.00  | root | data:TableScan_10                                                           |
+	//|   │ └─TableScan_10       | 1.00  | cop  | table:t1, range:[-inf,+inf], keep order:false, stats:pseudo                 |
+	//|   └─IndexReader_8        | 0.00  | root | index:IndexScan_7                                                           |
+	//|     └─IndexScan_7        | 0.00  | cop  | table:t2, index:c, range: decided by [t1.b], keep order:false, stats:pseudo |
```

Comment:
This kind of comment is ugly and hard to maintain. We may change the behavior of planner in the future, and then this comment is outdated, but we can not be aware to change this comment.

Maybe we should remove it. Or we can add this test in the integration test, not in this unit test:
```sql
-- first check the execution plan.
explain explain select /*+ TIDB_INLJ(t1,t2) */ t1.a from t t1 join t t2 on t1.b = t2.c;

-- then check the execution result.
select /*+ TIDB_INLJ(t1,t2) */ t1.a from t t1 join t t2 on t1.b = t2.c;
```